### PR TITLE
Add data source for sync

### PR DESCRIPTION
**Pull request by jtsaito**
This PR is for reviewing only.

- https://github.com/babbel/terraform-provider-travisci/pull/15
- https://github.com/babbel/terraform-provider-travisci/pull/16
- https://github.com/babbel/terraform-provider-travisci/pull/18
- https://github.com/babbel/terraform-provider-travisci/pull/19 <- you are here
- https://github.com/babbel/terraform-provider-travisci/pull/20

### CODE REVIEW

File: travisci/data_source_travisci_sync.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+	if localSyncedAt.Before(remoteSyncedAt) {
+		s, err := dataSourceSyncAndWaitForUpdate(client)
+		if err != nil {
+			return nil
+		}
+		d.Set("synced_at", s)
+	}
+
+	d.SetId("static")
```
Comment (jtsaito) [inline travisci/data_source_travisci_sync.go]: I left this as "static" string for now. @jansiwy suggested to transform this into a date.

File: travisci/data_source_travisci_sync.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+func dataSourceSyncAndWaitForUpdate(client *travis.Client) (string, error) {
+	_, err := client.Users.Sync()
+	if err != nil {
+		// See https://docs.travis-ci.com/api/#users
+		if !strings.Contains(err.Error(), "409") {
+			return "", err
+		}
+	}
+
+	for waitSeconds := 10 * time.Second; waitSeconds <= 60*time.Second; waitSeconds += 10 * time.Second {
```
Comment (jansiwy) [inline travisci/data_source_travisci_sync.go]: `60*time.Second` => `60 * time.Second`

Comment (jansiwy) [reply travisci/data_source_travisci_sync.go]: please extract thoese magic times into constants

File: travisci/data_source_travisci_sync.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+	d.SetId("static")
+
+	return nil
+}
+
+func dataSourceSyncAndWaitForUpdate(client *travis.Client) (string, error) {
+	_, err := client.Users.Sync()
+	if err != nil {
+		// See https://docs.travis-ci.com/api/#users
```
Comment (jansiwy) [inline travisci/data_source_travisci_sync.go]: well, just add a comment explaining that a 409 means that it's already syncing

**Comment by jtsaito** [comment]
closing review branch: outdated

**Pull request closed without merging**